OpenSim/Simulation/Model/ModelComponent.cpp

```diff
@@ -49,16 +49,18 @@ ModelComponent::ModelComponent(SimTK::Xml::Element& element)
 const Model& ModelComponent::getModel() const
 {
     if(!_model)
-        throw Exception("ModelComponent::getModel(): component does not "
-                "belong to a model. Have you called Model::initSystem()?"); 
+        throw Exception(getConcreteClassName() + "::getModel(): component '"
+            + getName() + "' does not belong to a model. "
+            "Have you called Model::initSystem()?"); 
     return *_model;
 }
 
 Model& ModelComponent::updModel()
 {
     if(!_model)
-        throw Exception("ModelComponent::updModel(): component does not "
-                "belong to a model. Have you called Model::initSystem()?"); 
+        throw Exception(getConcreteClassName() + "::updModel(): component '"
+            + getName() + "' does not belong to a model. "
+            "Have you called Model::initSystem()?");
```

Comment:
+1. This new message is slightly confusing (b/c updModel() method is still in ModelComponent; i), but is definitely more useful. A less confusing message would be "ModelComponent::updModel(): component getName() of type getConcreteClassName() does not belong to a model"